### PR TITLE
PackageJson: Prettify markdown/mdx on commit with lint-staged

### PR DESCRIPTION
**Pull request by connorlindsey**
Adds markdown and mdx files to lint-staged so that markdown files are formatted consistently. This will prevent extraneous formatting changes on future PRs to the docs. This will also format mdx files which will primarily impact Grafana UI/Storybook.

Ran `prettier --write docs/sources` and `prettier --check docs/sources` to format docs and verify prettier ran correctly.

### CODE REVIEW

**Review by torkelo — APPROVED** [review]

**Comment by torkelo** [comment]
Think it would be a good idea to do this on 8.1.x branch as well to make sure there are no conflicts when backport if doc PRS

File: package.json [hunk 1]
```diff
@@ -63,6 +63,9 @@
     ],
     "*pkg/**/*.go": [
       "gofmt -w -s"
+    ],
+    "*.{md,mdx}": [
```
Comment (zoltanbedi) [inline package.json:67]: would it make sense to add these extensions to line 61?

**Review by zoltanbedi — APPROVED** [review]
This is awesome!

**Review by osg-grafana — APPROVED** [review]
Much appreciated!

**Comment by osg-grafana** [comment]
This PR only touches Grafana and not GE* products. How do we expand this to touch everything?

**Comment by connorlindsey** [comment]
@osg-grafana Which other repos have docs that we should update?

**Comment by osg-grafana** [comment]
> @osg-grafana Which other repos have docs that we should update?

Responding via internal Slack.

**Pull request merged**